src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -203,6 +204,48 @@ class LambdaTest extends ExternalServicesTestUtils with MockitoSugar {
       )
     }
 
+  "handleRequest" should "send a slack message for each of the 3 folders' changes (title, description, both) if each folder's " +
+    "title/description is different from what's DDB" in {
+
+      val entityWithAnOldTitle = IO(structuralObjects(0).map(_.copy(title = Some("mock title_old_1"))))
+      val entityWithAnOldDescription =
+        IO(structuralObjects(1).map(_.copy(description = Some("mock description_old_1_1"))))
+      val entityWithAnOldTitleAndDescription = IO {
+        structuralObjects(2).map(
+          _.copy(title = Some("mock title_old_1_1_1"), description = Some("mock description_old_1_1_1"))
+        )
+      }
+
+      val mockLambda =
+        MockLambda(
+          convertFolderIdsAndRowsToListOfIoRows(folderIdsAndRows),
+          entitiesWithSourceIdReturnValue =
+            List(entityWithAnOldTitle, entityWithAnOldDescription, entityWithAnOldTitleAndDescription)
+        )
+
+      mockLambda.handleRequest(mockInputStream, mockOutputStream, mockContext)
+
+      val sentMessages = mockLambda.eventBridgeMessageCaptors.getAllValues.asScala.map(_.slackMessage)
```

Comment:
Do you think that some of this could be moved to the `verifyInvocationsAndArgumentsPassed` method of the `ExternalServicesTestUtils.scala` (with a new parameter called something like `sentUpdateSlackMessages: List[String] = Nil` and then add similar code to what's on line 290 of `ExternalServicesTestUtils.scala` to check it)?

I was thinking:

1.  it would ensure this check applies to all tests, assuring us that it is only invoked when there is an update  to be made 
2. in case this test is edited/removed on purpose/mistake? 

```
      mockLambda.verifyInvocationsAndArgumentsPassed(
        folderIdsAndRows,
        3,
        3,
        List(expectedMessageOne, expectedMessageTwo, expectedMessageThree)
      )
```

**In fact,** if you do that, then you could merge this test with the one above to reduce repetition.

Reply:
Yep, makes sense. I've added a section into `verifyInvocationsAndArgumentsPassed` which checks the slack message if there is one. I've removed the test I added as well.